recipes/asmtk/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir, apply_conandata_patches, export_conandata_patches
+import os
+
+required_conan_version = ">=2.1"
+
+class AsmjitConan(ConanFile):
+    name = "asmtk"
+    description = "AsmTK provides concepts that are useful mostly in AOT code-generation."
+    license = "Zlib"
+    topics = ("asmjit", "compiler", "assembler", "jit", "asmtk")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://asmjit.com"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    implements = ["auto_shared_fpic"]
+
+    def requirements(self):
+        # INFO: asmtk/globals.h: #include <asmjit/core.h>
+        # INFO asmtk consumes asmjit directly
+        self.requires("asmjit/cci.20240531", transitive_headers=True, transitive_libs=True)
```

Comment:
It will fail without transitive libraries. The asmtk requires asmtik directly in its public interface. Locally I had the follow error without transitive libs:

```
Run Build Command(s): /usr/bin/ninja -v -j12
[1/2] /usr/bin/c++  -isystem /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/include -isystem /home/uilian/.conan2/p/asmji041b298d31314/p/include -m64 -O3 -DNDEBUG -std=gnu++17 -MD -MT CMakeFiles/test_package.dir/test_package.cpp.o -MF CMakeFiles/test_package.dir/test_package.cpp.o.d -o CMakeFiles/test_package.dir/test_package.cpp.o -c /home/uilian/Development/conan/conan-center-index/recipes/asmtk/all/test_package/test_package.cpp
[2/2] : && /usr/bin/c++ -m64 -O3 -DNDEBUG -m64 CMakeFiles/test_package.dir/test_package.cpp.o -o test_package -L/home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib   -L/home/uilian/.conan2/p/asmji041b298d31314/p/lib -Wl,-rpath,/home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib:/home/uilian/.conan2/p/asmji041b298d31314/p/lib  /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so  -lpthread  -lrt  -lm && :
FAILED: test_package 
: && /usr/bin/c++ -m64 -O3 -DNDEBUG -m64 CMakeFiles/test_package.dir/test_package.cpp.o -o test_package -L/home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib   -L/home/uilian/.conan2/p/asmji041b298d31314/p/lib -Wl,-rpath,/home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib:/home/uilian/.conan2/p/asmji041b298d31314/p/lib  /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so  -lpthread  -lrt  -lm && :
/usr/bin/ld: CMakeFiles/test_package.dir/test_package.cpp.o: in function `main':
test_package.cpp:(.text.startup+0x44): undefined reference to `asmjit::_abi_1_13::CodeHolder::CodeHolder(asmjit::_abi_1_13::Support::Temporary const*)'
/usr/bin/ld: test_package.cpp:(.text.startup+0x58): undefined reference to `asmjit::_abi_1_13::CodeHolder::init(asmjit::_abi_1_13::Environment const&, unsigned long)'
/usr/bin/ld: test_package.cpp:(.text.startup+0x63): undefined reference to `asmjit::_abi_1_13::x86::Assembler::Assembler(asmjit::_abi_1_13::CodeHolder*)'
/usr/bin/ld: test_package.cpp:(.text.startup+0x94): undefined reference to `asmjit::_abi_1_13::x86::Assembler::~Assembler()'
/usr/bin/ld: test_package.cpp:(.text.startup+0x9c): undefined reference to `asmjit::_abi_1_13::CodeHolder::~CodeHolder()'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::String::reset()'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::InstAPI::stringToInstId(asmjit::_abi_1_13::Arch, char const*, unsigned long)'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::String::_opString(asmjit::_abi_1_13::String::ModifyOp, char const*, unsigned long)'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::InstAPI::validate(asmjit::_abi_1_13::Arch, asmjit::_abi_1_13::BaseInst const&, asmjit::_abi_1_13::Operand_ const*, unsigned long, asmjit::_abi_1_13::ValidationFlags)'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::BaseEmitter::labelByName(char const*, unsigned long, unsigned int)'
/usr/bin/ld: /home/uilian/.conan2/p/b/asmtkcb62b67b063b3/p/lib/libasmtk.so: undefined reference to `asmjit::_abi_1_13::_archTraits'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.


Traceback (most recent call last):
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/errors.py", line 52, in conanfile_exception_formatter
    yield
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/client/conanfile/build.py", line 14, in run_build_method
    conanfile.build()
  File "/home/uilian/Development/conan/conan-center-index/recipes/asmtk/all/test_package/conanfile.py", line 20, in build
    cmake.build()
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/tools/cmake/cmake.py", line 184, in build
    self._build(build_type, target, cli_args, build_tool_args, stdout=stdout, stderr=stderr)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/tools/cmake/cmake.py", line 163, in _build
    self._conanfile.run(command, env=env, stdout=stdout, stderr=stderr)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/model/conan_file.py", line 366, in run
    raise ConanException("Error %d while executing" % retcode)
conans.errors.ConanException: Error 1 while executing

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/cli/cli.py", line 193, in run
    command.run(self._conan_api, args[0][1:])
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/cli/command.py", line 171, in run
    info = self._method(conan_api, parser, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/cli/commands/create.py", line 140, in create
    run_test(conan_api, test_conanfile_path, ref, profile_host, profile_build, remotes, lockfile,
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/cli/commands/test.py", line 82, in run_test
    test_package(conan_api, deps_graph, path)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/cli/commands/create.py", line 186, in test_package
    conan_api.local.build(conanfile)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/api/subapi/local.py", line 103, in build
    run_build_method(conanfile, app.hook_manager)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/client/conanfile/build.py", line 12, in run_build_method
    with conanfile_exception_formatter(conanfile, "build"):
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/errors.py", line 67, in conanfile_exception_formatter
    _raise_conanfile_exc(exc)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/errors.py", line 46, in _raise_conanfile_exc
    raise ConanException(traceback.format_exc())
conans.errors.ConanException: Traceback (most recent call last):
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/errors.py", line 52, in conanfile_exception_formatter
    yield
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/client/conanfile/build.py", line 14, in run_build_method
    conanfile.build()
  File "/home/uilian/Development/conan/conan-center-index/recipes/asmtk/all/test_package/conanfile.py", line 20, in build
    cmake.build()
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/tools/cmake/cmake.py", line 184, in build
    self._build(build_type, target, cli_args, build_tool_args, stdout=stdout, stderr=stderr)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conan/tools/cmake/cmake.py", line 163, in _build
    self._conanfile.run(command, env=env, stdout=stdout, stderr=stderr)
  File "/opt/pyenv/versions/3.12.4/lib/python3.12/site-packages/conans/model/conan_file.py", line 366, in run
    raise ConanException("Error %d while executing" % retcode)
conans.errors.ConanException: Error 1 while executing

```

Reply:
Fixed it here https://github.com/conan-io/conan-center-index/pull/25528/commits/5e7a65aa556537b648dfa1786def27d37a0a2732

it is semantics - the consumer project is directly requiring this library, so we need to apply "include what you use" / "link what you use" principles - and list that on the consumer side